src/libflake/flake/flake.cc

```diff
@@ -823,20 +826,63 @@ void initLib(const Settings & settings)
 {
     auto prim_getFlake = [&settings](EvalState & state, const PosIdx pos, Value * * args, Value & v)
     {
-        std::string flakeRefS(state.forceStringNoCtx(*args[0], pos, "while evaluating the argument passed to builtins.getFlake"));
-        auto flakeRef = parseFlakeRef(state.fetchSettings, flakeRefS, {}, true);
-        if (state.settings.pureEval && !flakeRef.input.isLocked())
-            throw Error("cannot call 'getFlake' on unlocked flake reference '%s', at %s (use --impure to override)", flakeRefS, state.positions[pos]);
-
-        callFlake(state,
-            lockFlake(settings, state, flakeRef,
-                LockFlags {
-                    .updateLockFile = false,
-                    .writeLockFile = false,
-                    .useRegistries = !state.settings.pureEval && settings.useRegistries,
-                    .allowUnlocked = !state.settings.pureEval,
-                }),
-            v);
+        state.forceValue(*args[0], pos);
+
+        LockFlags lockFlags {
+            .updateLockFile = false,
+            .writeLockFile = false,
+            .warnModifiedLockFile = false,
+            .useRegistries = !state.settings.pureEval && settings.useRegistries,
+            .allowUnlocked = !state.settings.pureEval,
+        };
+
+        auto flakeRef =
+            args[0]->type() == nString
+            ? ({
+                std::string flakeRefS(state.forceStringNoCtx(*args[0], pos, "while evaluating the argument passed to builtins.getFlake"));
+                auto flakeRef = parseFlakeRef(state.fetchSettings, flakeRefS, {}, true);
+                if (state.settings.pureEval && !flakeRef.input.isLocked())
+                    throw Error("cannot call 'getFlake' on unlocked flake reference '%s', at %s (use --impure to override)", flakeRefS, state.positions[pos]);
+                flakeRef;
+            })
+            : ({
+                auto flakeInput = parseFlakeInput(state, std::nullopt, args[0], pos, {}, {});
+
+                /* Convert the result of parseFlakeInput() into a
+                   overrides map and a top-level flakeref. */
+                std::function<void(const InputPath & inputPath, const FlakeInput & input)> recurse;
+
+                recurse = [&](const InputPath & inputPath, const FlakeInput & input)
+                {
+                    if (!input.ref)
+                        state.error<EvalError>("'builtins.getFlake' requires attribute 'url'")
+                            .atPos(*args[0])
+                            .debugThrow();
+                    if (input.follows)
+                        state.error<EvalError>("'builtins.getFlake' does not permit attribute 'follows'")
+                            .atPos(*args[0])
+                            .debugThrow();
+                    if (!input.isFlake)
+                        state.error<EvalError>("'builtins.getFlake' does not permit attribute 'flake = false'; use 'builtins.fetchTree' instead")
```

Comment:
I don't think the user can use `fetchTree` for this, when they're _overriding_ a non-flake input.
(It does make sense for the top-level argument of `getFlake`, but not for the `inputs`.

Or is the goal that an arbitrary value can be passed via `inputs`? That would be great for testing with `checks`, but not a reason to reject non-flake `inputs` overrides.